### ISSUE TITLE
[3.4] Imagelist fields causing exception in overview

### ISSUE THREAD

**Issue by GawainLynch** [bolt/bolt#7002]
Details
-------

 - Relevant Bolt Version: 3.4

Reproduction
------------
```yaml
koalas:
    name: Koalas
    singular_name: Koalas
    fields:
        title:
            type: text
            class: large
            group: content
        slug:
            type: slug
            uses: title
        image:
            type: imagelist
        body:
            type: html
            height: 300px
    recordsperpage: 10
    icon_one: fa:leaf
    icon_many: fa:leaf
```

![image](https://user-images.githubusercontent.com/1427081/30469263-aa6a1a68-99ef-11e7-93b3-56521b5dcc9e.png)

**Comment by GawainLynch** [comment]
OK … so … this one is going to bite us hard I think.

What is happening is the listings are now using the new storage layer, so the magic methods will catch the `getImage` in `{% set listimage = content.getImage %}` and as there is a field named "image" it correctly (IMHO) returns the image field content instead of forwarding to the legacy trait method … Wasn't there people ranting about SoC in a recent team meeting :stuck_out_tongue_winking_eye: 

Thoughts on moving forward here?

**Comment by GawainLynch** [comment]
Addressed in #7044
